src/NodaTime/HighPerformance/Duration64.cs

```diff
@@ -0,0 +1,435 @@
+    /// </item>
+    /// </list>
+    /// </returns>
+    public int CompareTo(Duration64 other) => Nanoseconds.CompareTo(other.Nanoseconds);
+
+    /// <summary>
+    /// Implementation of <see cref="IComparable.CompareTo"/> to compare two durations.
+    /// See the type documentation for a description of ordering semantics.
+    /// </summary>
+    /// <remarks>
+    /// This uses explicit interface implementation to avoid it being called accidentally. The generic implementation should usually be preferred.
+    /// </remarks>
+    /// <exception cref="ArgumentException"><paramref name="obj"/> is non-null but does not refer to an instance of <see cref="Duration64"/>.</exception>
+    /// <param name="obj">The object to compare this value with.</param>
+    /// <returns>The result of comparing this instant with another one; see <see cref="CompareTo(NodaTime.HighPerformance.Duration64)"/> for general details.
+    /// If <paramref name="obj"/> is null, this method returns a value greater than 0.
+    /// </returns>
+    int IComparable.CompareTo(object? obj)
+    {
+        if (obj is null)
+        {
+            return 1;
+        }
+        Preconditions.CheckArgument(obj is Duration64, nameof(obj), "Object must be of type NodaTime.HighPerformance.Duration64.");
+        return CompareTo((Duration64) obj);
+    }
+    #endregion
+
+    #region IEquatable<Duration64> Members
+    /// <summary>
+    /// Indicates whether the current object is equal to another object of the same type.
+    /// See the type documentation for a description of equality semantics.
+    /// </summary>
+    /// <param name="other">An object to compare with this object.</param>
+    /// <returns>
+    /// true if the current object is equal to the <paramref name="other"/> parameter;
+    /// otherwise, false.
+    /// </returns>
+    public bool Equals(Duration64 other) => this == other;
+    #endregion
+
+    /// <summary>
+    /// Returns a <see cref="Duration"/> that represents the same number of ticks as the
+    /// given <see cref="TimeSpan"/>.
+    /// </summary>
+    /// <param name="timeSpan">The TimeSpan value to convert</param>
+    /// <exception cref="OverflowException">The timespan cannot be represented as a <see cref="Duration64"/>.</exception>
+    /// <returns>A new Duration with the same number of ticks as the given TimeSpan.</returns>
+    public static Duration64 FromTimeSpan(TimeSpan timeSpan) => FromTicks(timeSpan.Ticks);
+
+    /// <summary>
+    /// Returns a <see cref="TimeSpan"/> that represents the same number of ticks as this
+    /// <see cref="Duration"/>.
+    /// </summary>
+    /// <remarks>
+    /// If the number of nanoseconds in a duration is not a whole number of ticks, it is truncated towards zero.
+    /// For example, durations in the range [-99ns, 99ns] would all count as 0 ticks.
+    /// </remarks>
+    /// <exception cref="OverflowException">The number of ticks cannot be represented a signed 64-bit integer.</exception>
```

Comment:
This one is impossible, right? `TimeSpan` has a larger range than `Duration64`, doesn't it?